scripts/ansible-cli

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -e
+
+if [ -n "$DEBUG" ]; then
+  set -x
+fi
+
+usage()
+{
+    echo 'You should provide the following env vars:'
+    echo ''
+    echo "$0"
+    echo '  * `(MODULE)`: Module to use. Default: `shell`'
+    echo '  * `(ANSIBLE_MODULE_ARGS)`: Ad-hoc command to use.'
+    echo '  * `(TARGET_PATTERN)`: which managed nodes or groups you want to execute against. Default: `all`'
+    echo '  * `(SSH_PRIVATE_KEY)`: SSH key to use to connect on servers'
+    echo '  * `(SSH_PRIVATE_KEYS)`: SSH key array to use to connect on servers. Example: ["PRIVATE_KEY","PRIVATE_KEY"]'
+    echo '  * `(BASTION_URL)`: [DEPRECATED] SSH URL of the bastion server. Example: `admin@myserver.com`'
+    echo '  * `(SSH_JUMP_URL)`: SSH ProxyJump URL used with `ssh ProxyJump`. Example: `user1@Bastion1,user2@Bastion2`'
+    echo '  * `(EXTRA_ANSIBLE_VARS)`: Ansible extra-vars, set additional variables, json dict format.'
+    echo '  * `(ANSIBLE_REMOTE_USER)`: Ansible remote user. Default: `admin`.'
+    echo '  * `(DEBUG)`: Run in debug mode'
+    echo ''
+    echo 'ec2.py vars:'
+    echo '  * `(AWS_INVENTORY)`: If the Amazon EC2 dynamic inventory need to be used or no, can be eiter `true`, `false` or `auto`. `auto` checks if `AWS_ACCESS_KEY_ID` is set or not. Default: `auto`.'
+    echo '  * `(AWS_ACCESS_KEY_ID)`: Used by Amazon EC2 dynamic inventory'
+    echo '  * `(AWS_SECRET_ACCESS_KEY)`: Used by Amazon EC2 dynamic inventory'
+    echo '  * `(EC2_VPC_DESTINATION_VARIABLE)`: Can be either `ip_address` for public ip address or `private_ip_address`, see [ec2.ini](https://github.com/ansible/ansible/blob/devel/contrib/inventory/ec2.ini). Default: `private_ip_address`.'
+    echo ''
+    echo 'azure_rm.py vars:'
+    echo '  * `(AZURE_INVENTORY)`: If the Azure dynamic inventory need to be used or no, can be eiter `true`, `false` or `auto`. `auto` checks if `AZURE_SUBSCRIPTION_ID` is set or not. Default: `auto`.'
+    echo '  * `(AZURE_SUBSCRIPTION_ID)`: Used by Azure dynamic inventory'
+    echo '  * `(AZURE_TENANT_ID)`: Used by Azure dynamic inventory'
+    echo '  * `(AZURE_CLIENT_ID)`: Used by Azure dynamic inventory'
+    echo '  * `(AZURE_SECRET)`: Used by Azure dynamic inventory'
+    echo '  * `(AZURE_USE_PRIVATE_IP)`: Can be either `True` or `False`, see [azure_rm.py](https://raw.githubusercontent.com/ansible/ansible/devel/contrib/inventory/azure_rm.py). Default: `True`.'
+    echo ''
+    exit 1
+}
+if [ -n "$1" ]; then usage; fi
+
+source ansible-common.sh
+
+# default
+export TARGET_PATTERN=${TARGET_PATTERN:-all}
+
+echo "######################## Running ansible cli"
+
+ansible --version
+ansible ${TARGET_PATTERN} -u ${ANSIBLE_REMOTE_USER} -m ${MODULE} ${EXTRA_ANSIBLE_ARGS}
```

Comment:
in the usage `$MODULE` is shell. But I can't see where the default value is set
I would also prefix those variables by ANSIBLE_
cause we don't know later which type of scripts we will put into the toolkit.
let's try to have a clear scope of variables name (same for target)

Also a thought @talset maybe an idea ?

To run ansible shell module you need  `-m shell` `-a "...something."`
So does it make sence to set `MODULE` variable and have to add some extra ansible file containing `-a .....` ?

Should we keep that way or just put everything in EXTRA_ANSIBLE_ARGS ?

To be clear on use-cases, I noticed 2 usage that we have in our pipelines.
Lot of commands using :
```
  -l SUBSET, --limit=SUBSET
                        further limit selected hosts to an additional pattern
```
For this one we could provide a variable dedicated to it : ANSIBLE_LIMIT_HOSTS.
If defined you prefix by `-l` and concat it with EXTRA_ANSIBLE_ARGS
It will make it easier to read into the pipeline.

The other usecase I saw today in our pipelines : Run ansible with local connection using the template module to render a jinja template. So no really need of target eg `ansible localhost --connection local -m template -a "src=${DIR}/code/app/etc/local.xml.j2 dest=${DIR}/code/app/etc/local.xml" -e @${DIR}/secrets/variable.yml`

Reply:
+1 to add a prefix `ANSIBLE_` for every vars

I'm not sure about adding more "extra" variable. It might be too confusing and Im sure user will only use "EXTRA_ANSIBLE_ARGS" for every arg like --limit ...

Maybe a connecton_local: true/false or just write more examples of bases EXTRA_ANSIBLE_ARGS often used.

So I would go for

```
ANSIBLE_TARGET_PATTERN: all
ANSIBLE_MODULE: 'shell -a "echo foo"'
EXTRA_ANSIBLE_ARGS: --connection local
```

Reply:
Actually, for the ansible module command, I use: 
```
ANSIBLE_MODULE_ARGS
This sets the default arguments to pass to the ansible adhoc binary if no -a is specified.
```
[[doc](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#envvar-ANSIBLE_MODULE_ARGS)]
It means we do not need to add a `-a my-command` since this variable is taking care of this part.

I would go for: 
```
ANSIBLE_TARGET_PATTERN: all
ANSIBLE_MODULE: 'shell'
ANSIBLE_EXTRA_ARGS: --connection local (notice the difference with EXTRA_ANSIBLE_ARGS in order to be consistent)
ANSIBLE_MODULE_ARGS: 'echo foo'
```

IMHO, it makes more sense to split module from its command. Since some module doesn't need to run a command (i.e `ping`). It can also be confusing for the user to write a command like `shell -a "my-command"` ? 

I agree for `ANSIBLE_LIMIT_HOSTS`.

Reply:
> It means we do not need to add a -a my-command since this variable is taking care of this part.

ho perfect, could we have it into the usage/readme
Just matter of making clear for our customers cause they might not digg into ansible doc

> notice the difference with EXTRA_ANSIBLE_ARGS in order to be consistent

fair point indeed.
Could you apply it also for https://github.com/cycloidio/docker-cycloid-toolkit/blob/develop/scripts/ansible-runner#L20-L21

It will be easier for a user to switch from ansible runner to cli configuration.

Just use the same thing we did for bastion ssh key for backward compatibility

```
    echo '  * `(BASTION_URL)`: [DEPRECATED] SSH URL of the bastion server. Example: `admin@myserver.com`'
export SSH_PRIVATE_KEY="${SSH_PRIVATE_KEY:-$BASTION_PRIVATE_KEY}"
```